From 3b41683ebefc87ff30dbd3f5bf4e7c4357cc558f Mon Sep 17 00:00:00 2001
From: Sridhar Kasukurthi <skasuk@codeaurora.org>
Date: Mon, 10 Jul 2017 16:46:45 +0800
Subject: [PATCH 108/296] Notification: Fix no icon shown for data limit
 notification

When USB connects and data warning limit are enabled, the icon
for data limit is not shown in status bar after data limit reaches.

The reason is that when data limit notification reaches, the total
number of notifications for package android becomes 4, then will
auto-group, the group summary will re-use the firstly enqueued
notification's channel and importance, USB Charging notification is
the firstly enqueued one whose channel has IMPORTANT_MIN importance,
SystemUI will not show the icon for this unimportant notification,
it's considered as ambient notification in SystemUI.

For notifications in package android, they have different icons and
importance, so will not auto-group them.
---
 services/core/java/com/android/server/notification/GroupHelper.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/notification/GroupHelper.java b/services/core/java/com/android/server/notification/GroupHelper.java
index ce805aadec7..4068c2190c6 100644
--- a/services/core/java/com/android/server/notification/GroupHelper.java
+++ b/services/core/java/com/android/server/notification/GroupHelper.java
@@ -53,6 +53,7 @@ public class GroupHelper {
                 // Not grouped by the app, add to the list of notifications for the app;
                 // send grouping update if app exceeds the autogrouping limit.
                 synchronized (mUngroupedNotifications) {
+                    if ("android".equals(sbn.getPackageName())) return;
                     Map<String, LinkedHashSet<String>> ungroupedNotificationsByUser
                             = mUngroupedNotifications.get(sbn.getUserId());
                     if (ungroupedNotificationsByUser == null) {
-- 
2.11.0

